[10.0][MIG] geoengine_base_geolocalize migration

benwillig opened the pull request:
Also moved the openstreetmap api call in specific module.

@lmignon Your review is welcome!

benwillig committed:
[ADD] Module which only provides a method to call the Openstreetmap API

benwillig committed:
[MIG] geoengine_base_geolocalize migration

JonathanNEMRY commented on base_geolocalize_openstreetmap/models/geolocalize_openstreetmap.py:
```diff
@@ -0,0 +1,45 @@
+        """Get the latitude and longitude by requesting "mapquestapi"
+        see http://open.mapquestapi.com/geocoding/
+        """
+        url = 'http://nominatim.openstreetmap.org/search'
```
as it is an openstreetmap class then could it make sens to externalize this one into a class attribute `_url` ?? (in order to reuser request into other call?)

benwillig replied:
Yes, you are right. I also thought about using a config parameter but I think it doesn't make sense as this URL will maybe never change.

JonathanNEMRY reviewed (approved):
Just a litle question otherwise seems ok

asaunier committed:
Add support of onchange requests to convert_to_record

lmarion-source committed:
changing the default SRID (correction of the previous one)

lmarion-source committed:
correction on comment for field.py, create tests for the implementation of localization using either point or lat/long

lmarion-source committed:
split the search for address using OSM + testing

lmarion-source committed:
[FIX] code format and pylint:

Squashed commit of the following:

commit 527e21551ce785168fdcacd85e949d04b7f4d439
Author: Lindsay <lindsay.marion@acsone.eu>
Date:   Tue Jun 2 14:47:52 2020 +0200

    fixing typo

commit dd9c3ecc2b870470228b8a65642730cf86b36734
Author: Lindsay <lindsay.marion@acsone.eu>
Date:   Tue Jun 2 14:42:47 2020 +0200

    fixing pylint travis

commit 7899f9b312da7c6023e6ddf3f007bfb6abe556db
Author: Lindsay <lindsay.marion@acsone.eu>
Date:   Tue Jun 2 14:05:25 2020 +0200

    fix travis flake8 linting

commit 039e8670aa7938876e30efd6307358dd3019c8a4
Author: Lindsay <lindsay.marion@acsone.eu>
Date:   Tue Jun 2 13:54:07 2020 +0200

    fix travis pylint

commit fcfb09e0e3bddcda01071967d63b4cbc22d7dc15
Author: Lindsay <lindsay.marion@acsone.eu>
Date:   Tue Jun 2 13:45:09 2020 +0200

    fixing travis tests

commit ff2b39e66ddd6d8b73cd0150f0da527d88186031
Author: Lindsay <lindsay.marion@acsone.eu>
Date:   Tue Jun 2 12:07:45 2020 +0200

    fix import

commit cdf43dc0b6af7925fa2ac38acddcb93f743daf69
Author: Lindsay <lindsay.marion@acsone.eu>
Date:   Tue Jun 2 11:47:21 2020 +0200

    add responses to travis for testing

commit 79da0ee02a3ac76d605066f7a5d69ff336fb2bb5
Author: Lindsay <lindsay.marion@acsone.eu>
Date:   Tue Jun 2 11:40:55 2020 +0200

    disable pylint warning W0622

commit d6eab49374d01cdff18f4a42c7e01ce0156f90dd
Author: Lindsay <lindsay.marion@acsone.eu>
Date:   Tue Jun 2 10:29:58 2020 +0200

    formatting for linters

lmignon reviewed (approved):
Works like a charm in production. Thank you @lmarion-source

lmignon commented:
/ocabot merge patch
